ibm/service/codeengine/resource_ibm_code_engine_job.go

```diff
@@ -0,0 +1,817 @@
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "1",
+				ValidateFunc: validate.InvokeValidator("ibm_code_engine_job", "scale_cpu_limit"),
+				Description:  "Optional amount of CPU set for the instance of the job. For valid values see [Supported memory and CPU combinations](https://cloud.ibm.com/docs/codeengine?topic=codeengine-mem-cpu-combo).",
+			},
+			"scale_ephemeral_storage_limit": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "400M",
+				ValidateFunc: validate.InvokeValidator("ibm_code_engine_job", "scale_ephemeral_storage_limit"),
+				Description:  "Optional amount of ephemeral storage to set for the instance of the job. The amount specified as ephemeral storage, must not exceed the amount of `scale_memory_limit`. The units for specifying ephemeral storage are Megabyte (M) or Gigabyte (G), whereas G and M are the shorthand expressions for GB and MB. For more information see [Units of measurement](https://cloud.ibm.com/docs/codeengine?topic=codeengine-mem-cpu-combo#unit-measurements).",
+			},
+			"scale_max_execution_time": &schema.Schema{
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     7200,
+				Description: "The maximum execution time in seconds for runs of the job. This property can only be specified if `run_mode` is `task`.",
+			},
+			"scale_memory_limit": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "4G",
+				ValidateFunc: validate.InvokeValidator("ibm_code_engine_job", "scale_memory_limit"),
+				Description:  "Optional amount of memory set for the instance of the job. For valid values see [Supported memory and CPU combinations](https://cloud.ibm.com/docs/codeengine?topic=codeengine-mem-cpu-combo). The units for specifying memory are Megabyte (M) or Gigabyte (G), whereas G and M are the shorthand expressions for GB and MB. For more information see [Units of measurement](https://cloud.ibm.com/docs/codeengine?topic=codeengine-mem-cpu-combo#unit-measurements).",
+			},
+			"scale_retry_limit": &schema.Schema{
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     3,
+				Description: "The number of times to rerun an instance of the job before the job is marked as failed. This property can only be specified if `run_mode` is `task`.",
+			},
+			"created_at": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date when the resource was created.",
+			},
+			"entity_tag": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The version of the job instance, which is used to achieve optimistic locking.",
+			},
+			"href": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "When you provision a new job,  a URL is created identifying the location of the instance.",
+			},
+			"id": &schema.Schema{
```

Comment:
Kindly rename it to `job_id`
Reason is given in app resource comments.
Kindly make respective changes to all the references in the resource and docs as well

Reply:
Changed the ID according to suggestion